### PR TITLE
Remove fuel explode, fix diesel and clean code

### PR DESCRIPTION
**Pull request by wrefgtzweve**
This PR fixes the ignition code that should've been removed a long time ago, fixes diesel which has been broken the entire time and removes the propane explosion.

### CODE REVIEW

File: lua/entities/ent_jack_aidfuel_diesel/init.lua [hunk 1]
```diff
@@ -0,0 +1,25 @@
+AddCSLuaFile( "cl_init.lua" )
+AddCSLuaFile( "shared.lua" )
+include( "shared.lua" )
+
+function ENT:Initialize()
+    self:SetModel( "models/props_lab/harddrive02.mdl" )
+    self:SetColor( Color( 31, 31, 31 ) )
```
Comment (sarahsturgeon) [inline lua/entities/ent_jack_aidfuel_diesel/init.lua:7]: nit: Prefer using a local `Color` instance or `SetColor4Part`

File: lua/entities/ent_jack_aidfuel_diesel/init.lua [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    self:PhysicsInit( SOLID_VPHYSICS )
+    self:SetMoveType( MOVETYPE_VPHYSICS )
+    self:SetSolid( SOLID_VPHYSICS )
+    self:DrawShadow( true )
+    local phys = self:GetPhysicsObject()
+
+    if phys:IsValid() then
+        phys:Wake()
+        phys:SetMass( 35 )
+    end
```
Comment (sarahsturgeon) [inline lua/entities/ent_jack_aidfuel_diesel/init.lua]: nit: spacing
```suggestion

    local phys = self:GetPhysicsObject()
    if phys:IsValid() then
        phys:Wake()
        phys:SetMass( 35 )
    end
```

File: lua/entities/ent_jack_aidfuel_propane/init.lua [hunk 3]
```diff
@@ -45,23 +31,4 @@ end
 
 function ENT:OnTakeDamage( dmginfo )
     self:TakePhysicsDamage( dmginfo )
-    self.StructuralIntegrity = self.StructuralIntegrity - dmginfo:GetDamage()
```
Comment (sarahsturgeon) [inline lua/entities/ent_jack_aidfuel_propane/init.lua:48]: This isn't used anywhere else?

File: lua/entities/ent_jack_aidfuel_kerosene/init.lua [hunk 4]
```diff
@@ -2,20 +2,6 @@ AddCSLuaFile( "cl_init.lua" )
...
-    ent:SetPos( SpawnPos )
-    ent:SetNWEntity( "Owner", ply )
-    ent:Spawn()
-    ent:Activate()
-    local effectdata = EffectData()
-    effectdata:SetEntity( ent )
-    util.Effect( "propspawn", effectdata )
-
-    return ent
-end
```
Comment (sarahsturgeon) [inline lua/entities/ent_jack_aidfuel_kerosene/init.lua:17]: Doesn't work, or just not needed?

File: lua/entities/ent_jack_aidfuel_gasoline/init.lua [hunk 5]
```diff
@@ -44,47 +30,4 @@ end
...
 function ENT:OnTakeDamage( dmginfo )
     self:TakePhysicsDamage( dmginfo )
-    self.StructuralIntegrity = self.StructuralIntegrity - dmginfo:GetDamage()
-
-    if self.StructuralIntegrity <= 0 then
-        self:Asplode()
-    end
-end
-
-function ENT:Asplode()
```
Comment (sarahsturgeon) [inline lua/entities/ent_jack_aidfuel_gasoline/init.lua:54]: Was this only called from OnTakeDamage?

**Review by sarahsturgeon — APPROVED** [review]
LGTM 👍 

Only nitpicks and double-check comments

File: lua/entities/ent_jack_aidfuel_diesel/init.lua (hunk 1 above)
Comment (wrefgtzweve) [reply lua/entities/ent_jack_aidfuel_diesel/init.lua:7]: It's only called on entity creation, it wont affect it much at all

Comment (sarahsturgeon) [reply lua/entities/ent_jack_aidfuel_diesel/init.lua:7]: That's why it's a `nit` :)

File: lua/entities/ent_jack_aidfuel_propane/init.lua (hunk 3 above)
Comment (wrefgtzweve) [reply lua/entities/ent_jack_aidfuel_propane/init.lua:48]: Nope, good catch though i found other stuff that defines it for no reason

File: lua/entities/ent_jack_aidfuel_kerosene/init.lua (hunk 4 above)
Comment (wrefgtzweve) [reply lua/entities/ent_jack_aidfuel_kerosene/init.lua:17]: Not needed, gmod will pretty much just do this for you.

**Pull request merged**